tasks/download.sh

```diff
@@ -35,8 +35,19 @@ verify-file() {
 }
 
 download() {
+  local tmp_file_name="pe-tmp-file"
+
   printf '%s\n' "Downloading: ${1}"
-  curl -s -f -L -o "$2" "$1"
+  tmp_file=$(mktemp -p /tmp/ "$tmp_file_name")
+  echo "Temporary file created at: ${tmp_file}"
+  curl -s -f -L -o ${tmp_file} "$1"
+  if [tar -tzf ${tmp_file} >/dev/null]; then
```

Comment:
Bash syntax: need spacing around ` [ ` and ` ] `. The opening `[` is like a command (in old-time Posix it was literally `/bin/[`), and the closing `]` is a separate command-line argument.

Also, use quotes around variable name for safety, in case it contains spaces (style guide).

```suggestion
  if [ tar -tzf "${tmp_file}" >/dev/null ]; then
```